test/fail_compilation/bug9631.d

```diff
@@ -0,0 +1,21 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/bug9631.d(20): Error: cannot implicitly convert expression `F()` of type `bug9631.T1!().F` to `bug9631.T2!().F`
+---
+*/
+
+template T1()
+{
+    struct F { }
+}
+
+template T2()
+{
+    struct F { }
+}
+
+void main()
+{
+    T2!().F x = T1!().F();
```

Comment:
Sorry @ntrel but this isn't passing coverage.  You'll have to add a test that tests the case when both `t1` and `t2` are the same.

Reply:
Nevermind, The coverage report looks weird, and I project coverage looks good, so let's merge this.